### PR TITLE
Fixed cmd/task/list panic; cmd/dashboard server env var; more validation

### LINKED ISSUE #361: term/dashboard: use FUNNEL_SERVER
```
export FUNNEL_SERVER=http://funnel.compbio.ohsu.edu
funnel dashboard
```

Should use the server URL defined by environment variable

### LINKED ISSUE #363: cmd/task/list: unhandled panic
```
./funnel task list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x72e404]

goroutine 1 [running]:
net/http.(*Request).WithContext(0x0, 0x1c97b60, 0xc420010250, 0x1c97b60)
	/usr/local/go/src/net/http/request.go:332 +0x44
github.com/ohsu-comp-bio/funnel/client.(*Client).ListTasks(0xc4204782a0, 0x7f47d9c1a050, 0xc420010250, 0xc4201e5c50, 0xc420010250, 0xc4203d8d01, 0x6)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/client/client.go:78 +0x221
github.com/ohsu-comp-bio/funnel/cmd/task.List(0xc42000a14e, 0x10, 0x13b9a22, 0x5, 0x0, 0x0, 0x0, 0x1c8d6a0, 0xc42000c018, 0xc4201e5d18, ...)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/cmd/task/list.go:30 +0x1b8
github.com/ohsu-comp-bio/funnel/cmd/task.newCommandHooks.func3(0xc420476d80, 0x1d50a30, 0x0, 0x0, 0x0, 0x0)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/cmd/task/task.go:64 +0xfc
github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra.(*Command).execute(0xc420476d80, 0x1d50a30, 0x0, 0x0, 0xc420476d80, 0x1d50a30)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra/command.go:698 +0x47b
github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1d286a0, 0x1d29e00, 0x0, 0x0)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra/command.go:783 +0x349
github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra.(*Command).Execute(0x1d286a0, 0xc420000340, 0xc420000340)
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/Users/buchanae/src/github.com/ohsu-comp-bio/funnel/main.go:10 +0x2d
```

### PR DESCRIPTION
**Pull request by adamstruck**
closes #363 
closes #361

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -12,27 +12,37 @@ import (
...
 
-	// Strip trailing slash. A quick and dirty fix.
-	re := regexp.MustCompile("/+$")
-	address = re.ReplaceAllString(address, "")
+	re := regexp.MustCompile("^(.+://)?(.[^/]+)(.+)?$")
+	endpoint := re.ReplaceAllString(address, "$1$2")
+
+	reScheme := regexp.MustCompile("^.+://")
+	switch reScheme.MatchString(endpoint) {
+	case true:
```
Comment (buchanae) [inline client/client.go]: `switch` for true/false is pretty weird. `if` is better.

File: cmd/termdash/tasksource.go [hunk 2]
```diff
@@ -27,7 +27,10 @@ type TaskSource struct {
 
 func NewTaskSource(tesHTTPServerAddress string, pageSize uint32) *TaskSource {
 	// init funnel http client
-	cli := client.NewClient(tesHTTPServerAddress)
+	cli, err := client.NewClient(tesHTTPServerAddress)
+	if err != nil {
+		panic(err)
```
Comment (buchanae) [inline cmd/termdash/tasksource.go]: How hard is it to eliminate this panic? Seems like you're fixing one but adding another.

Comment (adamstruck) [reply cmd/termdash/tasksource.go]: I was able to eliminate all of the panics in this code in the latest commit.

File: client/client.go (hunk 1 above)
Comment (adamstruck) [reply client/client.go]: Good point.

**Comment by adamstruck** [comment]
latest commit make view ignore case

**Pull request merged**